### DISCUSSION TITLE
Best way to create playlists with NAS files

### DISCUSSION THREAD

**Discussion by Bachigaloupe** [music-assistant/support#661] [Q&A] [answered]
If this isn’t the correct place to ask, please have a Mod delete. 

I am so very thrilled to have recovered old mp3 files and stored them on my NAS. 

I mount the NAS in HA through a shell command and startup automation. 

Would someone point me to the creating playlists for dummies. I have tried reading up or Google search on m3u and other playlist formats. 

I can create a playlist and export it from iTunes just to have it show up in MA but it will be empty and I can add individual files using the MA add to playlist but this is not so time friendly. 

I really really love playing back my music on MA through an Arylic UpStream Amp and use the Aqara Magic Cube to control some of the media functions. 

I have done a search for similar discussions and sometimes it isn’t easy knowing how or where to search. 

I was hoping to see examples of how others that like to show off use MA for playlists on a server of mp3s and how to make custom playlists 

so is there some documentation area that I am missing or does everyone know how to create the best type and properly configured m3u files? Is m3u what I should use?

I am only a few months in to Home Assistant, the music helps when I am trying to figure out a complex (for me) automation. 

I appreciate your time reading this and if you can point me in the correct direction I would greatly appreciate it!

**Comment by marcelveldt** [comment, marked as answer]
The easiest way is to just use MA to edit your playlists.
The only downside is that MA is not yet capable of creating new playlists but you can edit them perfectly fine.

So create an empty file with the m3u extension in your music folder and you have a playlist in MA where you can add music to.

**Reply by Bachigaloupe** [reply]
Thank you and thanks again for your efforts!

It amazes me that I actually did exactly what you suggested without any documentation or what I mean is not knowing where to find answers as HA is still only 3 months old to me. 

I just wasn't sure if there was a Discord group where I could learn from others. 

I guess my hope was I would learn the secrets of the content of the Playlist file and edit it. 

Example = Adding files to playlists in MA is like using the UI in creating automations. 

My thought was some brilliant people probably edit the actual file which maybe code, I assumed was mostly text with some file names and paths. 

Thanks again Marcel!

Oh, please close this out and/or tell me where I should have posted.

**Reply by gNeeUs** [reply]
> The easiest way is to just use MA to edit your playlists. The only downside is that MA is not yet capable of creating new playlists but you can edit them perfectly fine.
> 
> So create an empty file with the m3u extension in your music folder and you have a playlist in MA where you can add music to.

Sorry I'm a bit late to this party, but now that we are already in 2024, I'm curious as to how one Edits a playlist. I have created a new playlist on Music Assistant, and when I go to a song or album in the Library I can add it to this playlist. However, from the playlist there seem to be no Edit options. I'm looking for the ability to Rename my playlist; create a playlist based on conditions such as Artist, Genre, Year, and other metadata; or drag and drop a variety of songs; or even create my own random playlist based on some conditions. I appreciate a pointer to instructions for this functionality (through the backend if needed). Thanks for a great product!

**Reply by OzGav** [reply]
You are quoting a reply from 2 years ago that is before version 2. Currently you can only add and delete from playlists.

**Reply by OzGav** [reply]
By the way that is why I closed this as it is outdated info. See the very last post

**Reply by gNeeUs** [reply]
> You are quoting a reply from 2 years ago that is before version 2. Currently you can only add and delete from playlists.

Understood! If there is a more up-to-date or appropriate forum, I'd be happy to repost, and look for enhancements to this service. Either way, I appreciate your support, cheers.

**Reply by OzGav** [reply]
Information about playlists is here https://music-assistant.io/usage/#playlists

Backlog item to enhance functionality is here https://github.com/orgs/music-assistant/projects/2/views/1?filterQuery=playlist&pane=issue&itemId=62585784

**Comment by Gobbo28** [comment]
Hello,

is there something special when creating the empty m3u file?
I created one inside my mounted nas folder but i cant add items to it. When i select a song and Click the „Add to Playlist“-Button nothing happens.
Am I missing something here?

Thanks in advance. :)

**Reply by Bachigaloupe** [reply]
> Hello,
> 
> 
> 
> is there something special when creating the empty m3u file?
> 
> I created one inside my mounted nas folder but i cant add items to it. When i select a song and Click the „Add to Playlist“-Button nothing happens.
> 
> Am I missing something here?
> 
> 
> 
> Thanks in advance. :)

I had this same thing happen and I think first you should try the refresh option on the empty playlist(s)

If that doesn't work then I created playlists in iTunes and exported them (empty) to my NAS music files directory and then could add them.

**Reply by Gobbo28** [reply]
Thank you for your suggestion. Unfortunatly that still doesnt work :(
Would it be possible that you send me a example playlist from you so that I can compare it to mine?

**Reply by marcelveldt** [reply]
please wait for the 2022.7 release as this will all be fixed, you can even create new playlists from the UI
If you want you can install the beta version, just enable beta versions in HACS

**Reply by Gobbo28** [reply]
I already installed 2022.7.9b which causes my complete music to disappear in MA.

**Reply by marcelveldt** [reply]
and you deselected the "favorites filter" at the top of the listing?

**Reply by Gobbo28** [reply]
yes, the content screen is just blank.

**Reply by Gobbo28** [reply]
Is there any possibility to debug music assistant?

**Reply by OzGav** [reply]
Just confirming that you have clicked on the heart ❤️ to make it hollow? Let us know if after doing that nothing is showing. A screenshot can be helpful.

**Reply by Gobbo28** [reply]
Thats correct. If I uncheck the Heart to Show all items, nothing will Show up. 
I can provide Screenshots later if you need.

**Reply by marcelveldt** [reply]
Strange! 
A few things to check:
1) Is everything setup correctly in the settings ? HA config --> devices and integrations --> Music Assistant --> Configure
2) Make sure that the path is set and the file provider is enabled
3) Check the Home Assistant logbook for errors/warning from music assistant. If in doubt share the homeassistant.log file here.
4) Please report what does work and what not. So are you not seeing any media at all or just some data missing, players visisble or not etc

**Reply by Gobbo28** [reply]
Ok I updated now to the 2022.7 Release and everything works perfect. 
Thank you for your help!

**Comment by snickers2k** [comment]
hey. i wonder why it's still not possible to simply add an m3u file via "add music provider" in UI ?
RadioBrowser is great. But adding a simple m3u playlist would be appreciated.

**Reply by OzGav** [reply]
You can but not through the UI. Put it on your NAS and it will be read when that is scanned

**Comment by OzGav** [comment]
Relates to v1
